### PR TITLE
New team/log thread

### PR DESCRIPTION
**Pull request by benirave**
Users are now able to create comments on a run's log.

- Created two endpoints under logs (Get Threads, Add a Comment)

I think I will still revise the method names and the comments, but the fundamentals are in place.

### CODE REVIEW

**Comment by PascalBoeschoten** [comment]
Looks like `CreateCommentDto` has decorators but `ThreadDto` is missing them (`@IsInt()`, etc.). Could these be added?

**Comment by CalvinHuynh** [comment]
Please create a new pull request to the develop branch. We do not merge anything directly into the master branch.

**Comment by CalvinHuynh** [comment]
Furthermore, this branch is not up to date with the develop branch. 
Please do the following: 

- [ ] Merge the develop branch into this branch 
- [ ] Fix the merge conflicts
- [ ] Do a general run/check to see if the newly added features are still working
- [ ] Create a PR if all of the above has been done into the develop branch.

In the future when you are creating new functionality, please base it on the latest develop branch.

**Comment by CalvinHuynh** [comment]
Furthermore, this branch is not up to date with the develop branch. 
Please do the following: 

- [ ] Merge the develop branch into this branch 
- [ ] Fix the merge conflicts
- [ ] Do a general run/check to see if the newly added features are still working
- [ ] Create a PR if all of the above has been done into the develop branch.

In the future when you are creating new functionality, please base it on the latest develop branch.

File: src/services/thread.service.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/*
```
Comment (CalvinHuynh) [inline src/services/thread.service.ts]: [Please merge develop into this branch and create a PR to the develop brach.](https://github.com/SoftwareForScience/jiskefet-api/pull/59#issuecomment-487412712)

File: src/entities/log.entity.ts [hunk 2]
```diff
@@ -126,4 +127,19 @@ export class Log {
     constructor(data: Log | {} = {}) {
         Object.assign(this, data);
     }
+
```
Comment (CalvinHuynh) [inline src/entities/log.entity.ts]: What is the reason to have the toTreadDto function inside the log entity class?

File: src/entities/log.entity.ts [hunk 3]
```diff
@@ -126,4 +127,19 @@ export class Log {
...
+    /**
+     * Method to convert Log to a ThreadDto
+     */
+    toThreadDto?(this: Log): ThreadDto {
```
Comment (CalvinHuynh) [inline src/entities/log.entity.ts]: What is the reason to have the toTreadDto function inside the log entity class?

File: src/controllers/log.controller.ts [hunk 4]
```diff
@@ -51,27 +69,75 @@ export class LogController {
...
+        return createResponseItem(getThreadById);
+    }
+
+    /**
+     * Creates a comment on the specific run's log and under the specific parent's log id
+     * @param createThreadDto Model to create a log
+     * - run Id refers to the run's log id
+     * - parent Id refers to the parent comment's id
+     */
+    @Post('/thread')
```
Comment (CalvinHuynh) [inline src/controllers/log.controller.ts]: Endpoint is not RESTful

File: src/controllers/log.controller.ts [hunk 5]
```diff
@@ -51,27 +69,75 @@ export class LogController {
...
+        } catch (error) {
+            return createErrorResponse(error);
+        }
+    }
+
+    /**
+     * Gets the full thread including the topic(run's log) by run's log id
+     * @param threadId Run's log id
+     */
+    @Get('/thread/:id')
```
Comment (CalvinHuynh) [inline src/controllers/log.controller.ts]: Endpoint is not RESTful

**Pull request closed without merging**